python/lsst/ts/ess/labjack/labjack_data_client.py

```diff
@@ -0,0 +1,321 @@
+        """
+        # Set of all LabJack channel names to read
+        channel_names: Set[str] = set()
+        for topic_info_dict in self.config.topics:
+            topic_info = types.SimpleNamespace(**topic_info_dict)
+            topic_name = topic_info.topic_name
+            topic_key = (topic_name, topic_info.sensor_name)
+
+            if topic_key in self.topic_handlers:
+                raise RuntimeError(
+                    f"topic_name={topic_name} with sensor_name={topic_info.sensor_name} "
+                    "listed more than once"
+                )
+            topic = getattr(self.topics, topic_name, None)
+            if topic is None:
+                raise RuntimeError(f"topic {topic_name} not found")
+            topic_handler = TopicHandler(
+                topic=topic,
+                sensor_name=topic_info.sensor_name,
+                field_name=topic_info.field_name,
+                location=topic_info.location,
+                channel_names=topic_info.channel_names,
+                offset=topic_info.offset,
+                scale=topic_info.scale,
+            )
+            self.topic_handlers[topic_key] = topic_handler
+            new_channel_names = set(topic_handler.channel_dict.values())
+            overlap_channel_names = new_channel_names & channel_names
+            if overlap_channel_names:
+                raise ValueError(
+                    f"Channel names {sorted(overlap_channel_names)} appear more than once."
+                )
+            channel_names |= new_channel_names
+        self.channel_names = tuple(sorted(channel_names))
+
+    def descr(self) -> str:
+        return f"identifier={self.config.identifier}"
+
+    async def connect(self) -> None:
+        """Connect to the LabJack and check we can read the specified channels.
+
+        Disconnect first, if connected.
+
+        Raises
+        ------
+        RuntimeError
+            If we cannot read the data.
+        """
+        loop = asyncio.get_running_loop()
+        with concurrent.futures.ThreadPoolExecutor(max_workers=1) as pool:
+            await asyncio.wait_for(
+                loop.run_in_executor(pool, self._blocking_connect()),
+                timeout=CONNECT_TIMEOUT,
+            )
+
+    async def disconnect(self):
+        """Disconnect from the LabJack. A no-op if disconnected."""
+        try:
+            loop = asyncio.get_running_loop()
+            with concurrent.futures.ThreadPoolExecutor(max_workers=1) as pool:
+                await asyncio.wait_for(
+                    loop.run_in_executor(pool, self._blocking_disconnect()),
+                    timeout=CONNECT_TIMEOUT,
+                )
+        finally:
+            self.handle = None
+
+    async def run(self) -> None:
+        """Read and process data from the LabJack."""
+        loop = asyncio.get_running_loop()
+        with concurrent.futures.ThreadPoolExecutor(max_workers=1) as pool:
```

Comment:
See above about using a single pool created in __init__, so thread is reused.

Reply:
I implemented the fix you suggested.